### DISCUSSION TITLE
Dynamic link to a Site in SiteCollection for usage in template

### DISCUSSION THREAD

**Discussion by TobiasK4T** [microsoft-search/pnp-modern-search#3162] [Q&A] [answered]
Hello everyone, I hope some can help because at this point I can't figure out how I get this working.

Site Setup:
Many SiteCollections connected through a Hub. Each Site has its own PnP Search Webpart which filters News (PromatedState:2). News do have an expiration date. So far so good this works all fine. 

Problem:
I want to include a Site on each Site Collection that shows "All News" related to the current Site Collection. Independent of expiration date. In the Search Results I want to link to this page with something like a "Show all" which works for all sites, so i can use a template.

![grafik](https://github.com/microsoft-search/pnp-modern-search/assets/99403575/b2aa8307-6a3d-40d4-b4d1-4a5463b51d3d)

This works fine with a static link like this:
`<p><a href="https://xxxx.sharepoint.com/sites/intranet/SitePages/Alle-News.aspx">Show all</a></p>`

But what I would like is something like the following, which I expected to work and would be super easy to implement, when every Site has an "Alle News.aspx" page:
`<p><a href="/SitePages/Alle-News.aspx">Show all</a></p>`

Current site I am on gets lost somehow:
currently on page similar to:
`https://xxxx.sharepoint.com/sites/intranet`

Resulting link:
`https://xxxx.sharepoint.com/SitePages/Alle-News.aspx`

![grafik](https://github.com/microsoft-search/pnp-modern-search/assets/99403575/ff9596bf-06be-4178-8222-008039d057f4)

Are there any workarounds? Is this a bug or expected behaviour? I don't want to change the URL by hand on over 50 sites for this, so I would greatly appreciate if someone has an idea or a hint.

**Comment by FranckyC** [comment, marked as answer]
Hi @TobiasK4T, in your template, you can do something like this using data context:

```
<p><a href="{{@root.context.site.absoluteUrl}}/SitePages/All-News.aspx">Show all</a></p>
```

**Reply by TobiasK4T** [reply]
Thanks this works. Tried so many variations, seems like this wasnt one of them.

Still is there a reason why this `<p><a href="/SitePages/Alle-News.aspx">Show all</a></p>` doesn't work?

**Reply by FranckyC** [reply]
That is the expected behavior actually, when you specify a relative link, it is always relative to the hostname. Your browser doesn't know about your SharePoint site URL :D

**Reply by TobiasK4T** [reply]
Alright, thank you very much for the fast response. :)
